components/SearchForJobs.tsx

```diff
@@ -10,55 +10,93 @@ import axios from "axios";
 import { Job } from "./Job";
 import { SwipeForJobs } from "./SwipeForJobs/SwipeForJobs";
 
+enum ViewState {
+  LOADING,
+  ERROR,
+  NO_DATA,
+  SUCCESS,
+}
+
 export const SearchForJobs = () => {
   const [jobs, onChangeJobs] = useState<string>("");
   const [location, onChangeLocation] = useState<string>("");
   const [data, setData] = useState<Job[]>([]);
-  const [isSubmitted, onChangeSubmit] = useState<boolean>(false);
+  const [viewState, setViewState] = useState<ViewState | null>(null);
 
   const handleSearch = (): void => {
+    setViewState(ViewState.LOADING);
     const url = `https://jobs.github.com/positions.json?description=${jobs}&location=${location}`;
     axios
       .get(url)
       .then((response) => {
+        !response.data.length
+          ? setViewState(ViewState.NO_DATA)
+          : setViewState(ViewState.SUCCESS);
+
         setData(response.data);
-        onChangeSubmit(true);
       })
-      .catch((error) => console.log(error));
+      .catch((error) => {
+        setViewState(ViewState.ERROR);
+        console.log(error);
+      });
   };
 
   const handleBack = (): void => {
-    onChangeSubmit(false);
+    setViewState(null);
+  };
+
+  const renderBack = (message?: string): React.ReactElement => {
+    return (
+      <View>
+        <Button onPress={handleBack} title="Back To Search" color="#841584" />
+        {message ? <p>{message}</p> : null}
+      </View>
+    );
+  };
+
+  const renderJobsView = (): React.ReactElement => {
+    switch (viewState) {
+      case ViewState.LOADING:
+        return <p>Loading...</p>;
+      case ViewState.ERROR:
+        return renderBack("Something went wrong.");
+      case ViewState.NO_DATA:
+        return renderBack("No jobs found. Blame Covid.");
+      case ViewState.SUCCESS:
+        return (
+          <View>
+            {data.length && (
```

Comment:
Don't understand what's going on here... so if data.length is not falsy (0), then it renders? I would have thought this statement evaluates to a boolean.

Reply:
If data.length is truthy, it renders the right block, else it renders nothing

Reply:
its a common short circuit technique. if the left side of && is truthy...execute right side. if left side of && is falsy...right side doesn't execute.